packages/common/src/entities/wallet.ts

```diff
@@ -15,4 +16,10 @@ export class Wallet {
     @ManyToOne(() => User, (user) => user.wallets)
     @JoinColumn({ name: "userId", referencedColumnName: "id" })
     user: User;
+
+    @OneToMany(() => Transaction, (transaction) => transaction.wallet_in)
+    transactions_in: Transaction[];
+
+    @OneToMany(() => Transaction, (transaction) => transaction.wallet_out)
+    transactions_out: Transaction[];
```

Comment:
I would just leave it as you had it before with a sinlge attribute `transactions` but just update the relationship to be a one to many. The disjunction relationship you have `||` represents what the ERD shows with the 2 relationships.

```suggestion
    @OneToMany(() => Transaction, (transaction) => transaction.wallet_in || transaction.wallet_out)
    transactions: Transaction[];
```